cdxev/set.py

```diff
@@ -223,11 +224,12 @@ def run(sbom: dict, updates: t.Sequence[dict[str, t.Any]], cfg: SetConfig) -> No
         try:
             target_list = ctx.component_map[update["id"][0]]
         except KeyError:
-            msg = LogMessage(
-                "Set not performed",
-                f'The component "{update["id"]}" was not found and could not be updated.',
-            )
-            raise AppError(log_msg=msg)
-
-        for target in target_list:
-            _do_update(target, update, ctx)
+            if not cfg.ignore_missing:
+                msg = LogMessage(
+                    "Set not performed",
+                    f'The component "{update["id"]}" was not found and could not be updated.',
+                )
+                raise AppError(log_msg=msg)
+        if "target_list" in locals():
+            for target in target_list:
+                _do_update(target, update, ctx)
```

Comment:
why not move the whole `for` into the `try`? This would result in the same logic and you would not be required to introduce this new `if`